programs/ssl/ssl_fork_server.c

```diff
@@ -100,13 +100,24 @@ int main( void )
     unsigned char buf[1024];
     const char *pers = "ssl_fork_server";
 
+#if defined(MBEDTLS_PLATFORM_C)
+    int is_parent = 1;
+    mbedtls_platform_context platform_ctx;
+#endif
     mbedtls_entropy_context entropy;
     mbedtls_ctr_drbg_context ctr_drbg;
     mbedtls_ssl_context ssl;
     mbedtls_ssl_config conf;
     mbedtls_x509_crt srvcert;
     mbedtls_pk_context pkey;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed!  mbedtls_platform_setup returned %d\n\n", ret );
+        goto exit;
+    }
+#endif
```

Comment:
for `ssl_fork_server`, I would expect this to be called once, for platform initialization, like you did, but I would also expect every child process to have access to the platform context. I would like to protect the context with a semaphore, and have the context sent as a shared resource.

Reply:
As discussed off line I raised an issue regarding this here: https://github.com/ARMmbed/mbedtls/issues/1583.